SafeExamBrowser.SystemComponents/VirtualMachineDetector.cs

```diff
@@ -97,69 +106,114 @@ private bool IsVirtualRegistry()
 		{
 			var isVirtualMachine = false;
 
-			// check historic hardware profiles
-			var hardwareConfKey = Microsoft.Win32.Registry.LocalMachine.OpenSubKey("SYSTEM\\HardwareConfig");
-			if (hardwareConfKey != null)
+			// the resulting IsVirtualRegistry() would be massive so split it
+			isVirtualMachine |= IsVirtualRegistryHardwareConfig();
+			isVirtualMachine |= IsVirtualRegistryDeviceCache();
+
+			return isVirtualMachine;
+		}
+
+
+		/// <summary>
+		/// Scans (historic) hardware configurations in the registry.
+		/// </summary>
+		private bool IsVirtualRegistryHardwareConfig()
+		{
+			bool isVirtualMachine = false;
+
+			/** 
+			 * scanned registry format:
+			 * 
+			 * HKLM\SYSTEM\HardwareConfig\{configId=uuid}\ComputerIds
+			 *	- {computerId=uuid}: {computerSummary=hardwareInfo}
+			 *	
+			 */
+			IEnumerable<string> hardwareConfigSubkeys;
+			const string hwConfigParentKey = "HKEY_LOCAL_MACHINE\\SYSTEM\\HardwareConfig";
+			if (!registry.TryGetSubKeys(hwConfigParentKey, out hardwareConfigSubkeys))
```

Comment:
Dito (brackets).

Reply:
Also, the out parameter declaration can be inline, i.e. `out var hardwareConfigurationSubKeys`.